### PR TITLE
Add agentless connection integration test

### PR DESCRIPTION
**Pull request by capnspacehook**
Follow up PR to #21953.

### CODE REVIEW

File: integration/integration_test.go [hunk 1]
```diff
@@ -7223,6 +7226,160 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
 	require.Equal(t, testFilePath, sftpEvent.GetString(events.SFTPPath))
 }
 
+func testAgentlessConnection(t *testing.T, suite *integrationTestSuite) {
+	// create Teleport instance
+	teleInst := suite.newTeleport(t, nil, true)
+	t.Cleanup(func() {
+		teleInst.StopAll()
```
Comment (codingllama) [inline integration/integration_test.go:7233]:
```suggestion
		_ = teleInst.StopAll()
```

so it's clear we are swallowing the error.

I've other folks do this too:

```suggestion
		require.NoError(teleInst.StopAll())
```

(No strong opinions on my part, as long as we pick one.)

File: integration/integration_test.go [hunk 2]
```diff
@@ -7223,6 +7226,160 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
...
+
+	var nodeCreated bool
+	for !nodeCreated {
+		select {
+		case e := <-w.Events():
+			if e.Type == types.OpPut {
+				nodeCreated = true
+			}
+		case <-w.Done():
+			require.Fail(t, "did not recieve node create event")
```
Comment (codingllama) [inline integration/integration_test.go:7317]: typo and nit:

```suggestion
			t.Fatal("Did not receive node create event")
```

File: integration/integration_test.go [hunk 3]
```diff
@@ -7223,6 +7226,160 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
...
+		Kinds: []types.WatchKind{
+			{
+				Kind: types.KindNode,
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	var nodeCreated bool
+	for !nodeCreated {
```
Comment (codingllama) [inline integration/integration_test.go:7310]:
```suggestion
	for nodeCreated := false; !nodeCreated; {
```

The smaller the scope, the better.

File: integration/integration_test.go [hunk 4]
```diff
@@ -7223,6 +7226,160 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
...
+			require.Fail(t, "did not recieve node create event")
+		}
+	}
+	require.NoError(t, w.Close())
+
+	// connect to node
+	proxyClient, err := tc.ConnectToProxy(ctx)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		proxyClient.Close()
```
Comment (codingllama) [inline integration/integration_test.go:7326]: Ditto for not swallowing errors silently.

File: integration/integration_test.go [hunk 5]
```diff
@@ -7223,6 +7226,160 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
...
+
+	lis, err := net.Listen("tcp", Loopback+":")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, lis.Close())
+	})
+
+	go func() {
+		nConn, err := lis.Accept()
+		require.NoError(t, err)
```
Comment (codingllama) [inline integration/integration_test.go:7369]: Let's defer or t.Cleanup close `nConn` and `conn`, they will leak if some requires fail.

**Review by codingllama — COMMENTED** [review]
Always nice to see test coverage follow ups, especially integration.

File: integration/integration_test.go (hunk 1 above)
Comment (capnspacehook) [reply integration/integration_test.go]: Done

File: integration/integration_test.go (hunk 2 above)
Comment (capnspacehook) [reply integration/integration_test.go]: Done

File: integration/integration_test.go (hunk 3 above)
Comment (capnspacehook) [reply integration/integration_test.go]: Good point, done

File: integration/integration_test.go (hunk 4 above)
Comment (capnspacehook) [reply integration/integration_test.go]: Done

File: integration/integration_test.go (hunk 5 above)
Comment (capnspacehook) [reply integration/integration_test.go:7368]: Good catch, done

**Review by codingllama — APPROVED** [review]

File: integration/integration_test.go [hunk 6]
```diff
@@ -7223,6 +7226,165 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
...
+	openSSHCA, err := authClient.GetCertAuthority(ctx, types.CertAuthID{
+		Type:       types.OpenSSHCA,
+		DomainName: helpers.Site,
+	}, false)
+	require.NoError(t, err)
+
+	caCheckers, err := sshutils.GetCheckers(openSSHCA)
+	require.NoError(t, err)
+
+	key, err := native.GeneratePrivateKey()
```
Comment (zmb3) [inline integration/integration_test.go:7255]: Don't we have a pool of keys we can pull from? I thought @jakule recently made some updates that reduced CPU usage and flakiness by dramatically reducing the amount of key generation we do.

Comment (capnspacehook) [reply integration/integration_test.go:7255]: He told me recently to do it this way, as the `native` package will precompute keys when tests are being run IIRC

File: integration/integration_test.go [hunk 7]
```diff
@@ -7223,6 +7226,165 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
 	require.Equal(t, testFilePath, sftpEvent.GetString(events.SFTPPath))
 }
 
+func testAgentlessConnection(t *testing.T, suite *integrationTestSuite) {
+	// create Teleport instance
+	teleInst := suite.newTeleport(t, nil, true)
+	t.Cleanup(func() {
```
[the remark this answers is not in the document]
Comment (capnspacehook) [reply integration/integration_test.go:7232]: Looks like `newTeleport` doesn't handle cleaning up at all. I agree with your guidelines, but I don't think addressing `newTeleport` behavior is in the scope of this PR.

I could address that in another PR though if you think that'd be best.

[the remark this answers is not in the document]
Comment (zmb3) [reply integration/integration_test.go:7232]: I agree with your assessment. Leave it out of scope for now.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**